packages/tgpu-cli/gen.mjs

```diff
@@ -1,64 +1,107 @@
 // @ts-check
 
 import fs from 'node:fs/promises';
+import path from 'node:path';
 import { ArrayInfo, StructInfo, TemplateInfo, WgslReflect } from 'wgsl_reflect';
 
 const cwd = new URL(`file:${process.cwd()}/`);
 
 const LENGTH_VAR = 'arrayLength';
 
+export class Context {
+  /**
+   * @param { {toTs: boolean, toCommonJs: boolean} } options
+   */
+  constructor(options) {
+    this.toTs = options.toTs;
+    this.toCommonJs = options.toCommonJs;
+
+    /** @type { string[] } */
+    this.exportsList = [];
+  }
+}
+
 /**
  * @param { string } input
  * @param { string } output
+ * @param { boolean } toTs
+ * @param { boolean } toCommonJs
+ * @param { boolean } overwriteExisting
  */
-async function main(input, output) {
+async function main(input, output, toTs, toCommonJs, overwriteExisting) {
   const inputPath = new URL(input, cwd);
   const outputPath = new URL(output, cwd);
-  const toTs = output.endsWith('.ts');
   const inputContents = await fs.readFile(inputPath, 'utf8');
 
-  await fs.writeFile(outputPath, generate(inputContents, toTs));
+  const generated = generate(inputContents, new Context({ toTs, toCommonJs }));
+
+  const fileExists = await fs
+    .access(output)
+    .then(() => true)
+    .catch(() => false);
+
+  if (fileExists && !overwriteExisting) {
+    console.warn(
+      `File ${output} already exists. Use --overwrite option to replace existing files.`,
+    );
+    return;
+  }
```

Comment:
This interacts with watch mode in a non-intuitive way. I think maybe watch mode should implicitly have the `--overwrite` flag?

Reply:
I'm afraid it's quite dangerous to just treat watch mode as having the overwrite flag. While overwriting is expected when editing source file, it isn't when we run the script. Imo there are 2 options: add an immediate warning when someone runs the script with `watch`, but with no `overwrite`; ooor in that case only check if we're overwriting the first time we generate a file, not when the file changes. what do you think?

Reply:
I think i like the second approach better. It is how I would assume `watch` works by default.

Reply:
Same here, second approach makes the most sense to me.

Reply:
I would still lean towards having both modes be explicit (either keep existing files, or overwrite existing files), and by default, exit with an error if a matching file already exists. In that way, running `watch` a second time will result in a proper error, not just a set of warnings for existing files. Then the user can choose whether they want to `--overwrite` or `--keep`.